src/overlays/actors/ovl_En_Zl1/z_en_zl1.c

```diff
@@ -15,51 +15,650 @@ void EnZl1_Destroy(Actor* thisx, GlobalContext* globalCtx);
+    Player* player = PLAYER;
+    EnZl1* this = THIS;
+
+    func_80038290(globalCtx, thisx, &this->unk_200, &this->unk_206, thisx->posRot2.pos);
+
+    if (this->unk_1E6 != 0) {
+        if (func_8002F334(thisx, globalCtx) != 0) {
+            this->unk_1E6 = 0;
+        }
+    } else {
+        if (func_8002F194(thisx, globalCtx)) {
+            this->unk_1E6 = 1;
+        } else {
+            if (thisx->posRot.pos.y <= player->actor.posRot.pos.y) {
+                func_8002F2F4(thisx, globalCtx);
+            }
+        }
+    }
+
+    Collider_CylinderUpdate(&this->actor, &this->collider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+}
+
+#ifdef NON_MATCHING
+// regalloc issues with the envCtx.unk_E2 section - using v0 instead of v1
+// silly if (rotDiff) line added because it fixes almost every regalloc issue
+void func_80B4B010(Actor* thisx, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    EnZl1* this = THIS;
+    AnimationHeader* animationHeader;
+    s32 pad;
+    Vec3f vec1 = D_80B4E630;
+    Vec3f vec2 = D_80B4E63C;
+    Vec3f playerPos = D_80B4E648;
+    s16 rotDiff;
+
+    if (func_8002F194(thisx, globalCtx)) {
+        animationHeader = &D_06010B38;
+        SkelAnime_ChangeAnim(&this->skelAnime, animationHeader, 1.0f, 0.0f,
+                             SkelAnime_GetFrameCount(&animationHeader->genericHeader), 3, -10.0f);
+        this->unk_1E8 = Gameplay_CreateSubCamera(globalCtx);
+        Gameplay_ChangeCameraStatus(globalCtx, 0, 1);
+        Gameplay_ChangeCameraStatus(globalCtx, this->unk_1E8, 7);
+        func_800C0808(globalCtx, this->unk_1E8, player, 0x21);
+        globalCtx->envCtx.unk_E2[0] = 0xFF;
+        globalCtx->envCtx.unk_E2[1] = 0xFF;
+        if (rotDiff) {} // fixes a lot of regalloc issues but is definitely a fake match improvement
+        globalCtx->envCtx.unk_E2[2] = 0xFF;
+        globalCtx->envCtx.unk_E2[3] = 0x18;
+        globalCtx->envCtx.unk_E1 = 1;
+        func_800C04D8(globalCtx, this->unk_1E8, &vec1, &vec2);
+        func_800C0704(globalCtx, this->unk_1E8, 30.0f);
+        func_800B3840(0x20);
+        Interface_ChangeAlpha(2);
+        player->actor.posRot.pos = playerPos;
+        player->actor.speedXZ = 0.0f;
+        this->unk_1E2 = 0;
+        this->actionFunc = func_80B4B240;
+        func_800F5C64(0x51);
+    } else {
+        rotDiff = ABS(thisx->rotTowardsLinkY - thisx->shape.rot.y);
+        if (rotDiff < 0x238E) {
+            if (player->actor.posRot.pos.y < thisx->posRot.pos.y) {
+                // nothing
+            } else {
+                func_8002F2F4(thisx, globalCtx);
+            }
+        }
+    }
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl1/func_80B4B010.s")
+#endif
+
+#ifdef NON_MATCHING
+// stack issue in cases that use msgCtx->choiceIndex - something is wonky with msgCtx
+/*void func_80B4B240(Actor* thisx, GlobalContext* globalCtx) {
```

Comment:
same here, i dont think this should be taking a thisx and it might help you with your problems
also if using non_matching no need to comment it out